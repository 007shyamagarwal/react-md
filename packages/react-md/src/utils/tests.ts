import * as React from "react";
import { create, ReactTestRendererJSON } from "react-test-renderer";

export function createSnapshot(children: React.ReactElement<any>): ReactTestRendererJSON | null {
  return create(children).toJSON();
}

export function expectSnapshot(children: React.ReactElement<any>): void {
  expect(createSnapshot(children)).toMatchSnapshot();
}
