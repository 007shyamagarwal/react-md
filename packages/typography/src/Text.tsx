import * as React from "react";
import * as PropTypes from "prop-types";
import cn from "classnames";
import {
  IBaseProps,
  FontWeights,
  TextAligns,
  TextDecorations,
} from "@react-md/internal-types";

export interface ITextChildrenFunction {
  className: string;
}

export type TextTypes = "headline-1" | "headline-2" | "headline-3" | "headline-4" | "headline-5" | "headline-6" | "subtitle-1" | "subtitle-2" | "body-1" | "body-2" | "caption" | "overline" | "button";

export interface ITextProps extends IBaseProps {
  tagName?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "p" | "span" | "div" | "button" | "caption";
  type?: TextTypes | null;
  weight?: FontWeights;
  decoration?: TextDecorations;
  align?: TextAligns;
  children?: React.ReactNode | ((props: ITextChildrenFunction) => React.ReactNode | null);
}

export interface ITextState {
  Component: React.ReactType | null;
}

/**
 * A utility function to get the html tag the Text component should render as.
 * By default, the component will attempt to render just as a text string if there
 * are no styling requirements provided. Otherwise, it will attempt to render as
 * the provided `tagName` or some auto-logic for determinine what html tag should be
 * used for styling.
 *
 * When the component should be rendered as text, null will be returned. Otherwise
 * the string html tag name will be returned.
 */
function getComponent(props: ITextProps): string | null {
  const {
    tagName,
    type,
    weight,
    decoration,
    align,
  } = props;

  if (tagName) {
    return tagName;
  } else if (type) {
    switch (type) {
      case "headline-1":
        return "h1";
      case "headline-2":
        return "h2";
      case "headline-3":
        return "h3";
      case "headline-4":
        return "h4";
      case "headline-5":
        return "h5";
      case "headline-6":
      case "subtitle-1":
      case "subtitle-2":
        return "h6";
      case "body-1":
      case "body-2":
        return "p";
      case "caption":
        return "caption";
      default:
        // button and overline
        return "span";
    }
  } else if (weight || decoration || align) {
    return "span";
  }

  return null;
}

export default class Text extends React.Component<ITextProps, ITextState> {
  public static propTypes = {
    style: PropTypes.object,
    className: PropTypes.string,
    tagName: PropTypes.oneOf([
      "h1",
      "h2",
      "h3",
      "h4",
      "h5",
      "h6",
      "p",
      "span",
      "div",
      "button",
      "caption",
    ]),
    type: PropTypes.oneOf([
      "headline-1",
      "headline-2",
      "headline-3",
      "headline-4",
      "headline-5",
      "headline-6",
      "subtitle-1",
      "subtitle-2",
      "body-1",
      "body-2",
      "caption",
      "overline",
      "button",
    ]),
    align: PropTypes.oneOf([
      "left",
      "center",
      "right",
      "justify",
      "inherit",
      "initial",
    ]),
    decoration: PropTypes.oneOf([
      "overline",
      "underline",
      "line-through",
      "none",
      "initial",
      "inherit",
    ]),
    weight: PropTypes.oneOf([
      "light",
      "regular",
      "medium",
      "semibold",
      "bold",
    ]),
    children: PropTypes.oneOfType([
      PropTypes.func,
      PropTypes.node,
    ]),
  };

  public static getDerivedStateFromProps(nextProps: ITextProps, prevState: ITextState): ITextState | null {
    const Component = getComponent(nextProps);
    if (prevState.Component !== Component) {
      return { Component };
    }

    return null;
  }

  public state = { Component: null };

  public render() {
    const { Component } = this.state;
    const {
      className: propClassName,
      children,
      tagName,
      type,
      weight,
      decoration,
      align,
      ...props
    } = this.props;

    const className = cn("md-typography", {
      [`md-typography--${align}`]: !!align,
      [`md-typography--${weight}`]: !!weight,
      [`md-typography--${decoration}`]: !!decoration,
      [`md-typography--${type}`]: !!type,
    }, propClassName);

    if (typeof children === "function") {
      return children({ className });
    } else if (!Component) {
      return children;
    }

    return React.createElement(Component, { ...props, className }, children);
  }
}
