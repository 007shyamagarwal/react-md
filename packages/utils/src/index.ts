export { default as positionRelativeTo } from "./positionRelativeTo";
export * from "./positionRelativeTo";
