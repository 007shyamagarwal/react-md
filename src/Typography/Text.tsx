import * as cn from "classnames";
import * as PropTypes from "prop-types";
import * as React from "react";
import {
  IBaseProps,
  FontWeights,
  TextAligns,
  TextDecorations,
} from "../types";

export interface ITextChildrenFunction {
  className: string;
}

export interface ITextProps extends IBaseProps {
  type?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "section" | "aside" | "caption" | "p" | "span" | null;
  weight?: FontWeights;
  decoration?: TextDecorations;
  align?: TextAligns;
  h?: 1 | 2 | 3 | 4 | 5 | 6 | null;
  p?: boolean | 1 | 2 | null;
  display?: 1 | 2 | 3 | 4 | null;
  headline?: boolean;
  title?: boolean;
  subheading?: boolean | 1 | 2 | null;
  caption?: boolean;
  children?: React.ReactNode | ((props: ITextChildrenFunction) => React.ReactNode | null);
}

export interface ITextState {
  Component: React.ReactType | null;
}

/**
 * A utility function to get the html tag the Text component should render as.
 * By default, the component will attempt to render just as a text string if there
 * are no styling requirements provided. Otherwise, it will attempt to render as
 * the provided `type` or some auto-logic for determinine what html tag should be
 * used for styling.
 *
 * When the component should be rendered as text, null will be returned. Otherwise
 * the string html tag name will be returned.
 */
function getComponent(props: ITextProps): string | null {
  const {
    type,
    weight,
    decoration,
    display,
    headline,
    title,
    subheading,
    caption,
    p,
    h,
    align,
  } = props;

  if (type) {
    return type;
  } else if (display || headline) {
    return "h1";
  } else if (title) {
    return "h2";
  } else if (subheading) {
    return subheading === true || subheading === 1 ? "h4" : "h3";
  } else if (p) {
    return p === true || p === 1 ? "p" : "aside";
  } else if (caption) {
    return "caption";
  } else if (h) {
    return `h${h}`;
  } else if (weight || decoration || align) {
    return "span";
  }

  return null;
}

export default class Text extends React.Component<ITextProps, ITextState> {
  public static propTypes = {
    style: PropTypes.object,
    className: PropTypes.string,
    type: PropTypes.oneOf([
      "h1",
      "h2",
      "h3",
      "h4",
      "h5",
      "h6",
      "section",
      "aside",
      "caption",
      "p",
      "span",
    ]),
    align: PropTypes.oneOf([
      "left",
      "center",
      "right",
      "justify",
      "inherit",
      "initial",
    ]),
    decoration: PropTypes.oneOf([
      "overline",
      "underline",
      "line-through",
      "none",
      "initial",
      "inherit",
    ]),
    weight: PropTypes.oneOf([
      "light",
      "regular",
      "medium",
      "semibold",
      "bold",
    ]),
    children: PropTypes.oneOfType([
      PropTypes.func,
      PropTypes.node,
    ]),
  };

  public static getDerivedStateFromProps(nextProps: ITextProps, prevState: ITextState): ITextState | null {
    const Component = getComponent(nextProps);
    if (prevState.Component !== Component) {
      return { Component };
    }

    return null;
  }

  public state = { Component: null };

  public render() {
    const { Component } = this.state;
    const {
      className: propClassName,
      children,
      type,
      weight,
      decoration,
      h,
      p,
      title,
      subheading,
      display,
      align,
      caption,
      headline,
      ...props,
    } = this.props;

    const className = cn("md-typography", {
      [`md-typography--${align}`]: !!align,
      [`md-typography--${weight}`]: !!weight,
      [`md-typography--${decoration}`]: !!decoration,
      [`md-typography--display-${display}`]: !!display,
      "md-typography--headline": headline,
      "md-typography--title": title,
      "md-typography--subheading-1": subheading === true || subheading === 1,
      "md-typography--subheading-2": subheading === 2,
      "md-typography--body-1": p === 1 || p === true,
      "md-typography--body-2": p === 2,
      "md-typography--caption": caption,
    }, propClassName);

    if (typeof children === "function") {
      return children({ className });
    } else if (!Component) {
      return children;
    }

    return React.createElement(Component, { ...props, className }, children);
  }
}
