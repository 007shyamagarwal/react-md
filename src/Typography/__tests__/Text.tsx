import * as React from 'react';
import { expectSnapshot } from '../../utils/tests';

import Text from '../Text';

const HELLO_WORLD = 'Hello, world!';
const LONG_TEXT = 'This is another string that is just some text. I\'m not sure how helpful this is though.';

describe('Text', () => {
  describe('rendering return value', () => {
    it('should render a text string when there are no additional props provided', () => {
      expectSnapshot(<Text>{HELLO_WORLD}</Text>);
      expectSnapshot(<Text>{LONG_TEXT}</Text>);
    });

    it('should render as the provided type prop', () => {
      expectSnapshot(<Text type="h1">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text type="h2">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text type="h3">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text type="h4">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text type="h5">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text type="h6">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text type="section">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text type="aside">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text type="caption">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text type="p">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text type="span">{HELLO_WORLD}</Text>);
    });

    it('should render as an h1 tag if the display prop is provided', () => {
      expectSnapshot(<Text display={1}>{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={2}>{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={3}>{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={4}>{HELLO_WORLD}</Text>);
    });

    it('should render as an h1 tag if the headline prop is enabled', () => {
      expectSnapshot(<Text headline={true}>{HELLO_WORLD}</Text>);
    });

    it('should render as the h2 tag if the title prop is enabled', () => {
      expectSnapshot(<Text title={true}>{HELLO_WORLD}</Text>);
    });

    it('should render as an h4 tag if the subheading prop is enabled or set to 1', () => {
      expectSnapshot(<Text subheading={true}>{HELLO_WORLD}</Text>);
      expectSnapshot(<Text subheading={1}>{HELLO_WORLD}</Text>);
    });

    it('should render as an h3 tag if the subheading prop is set to 2', () => {
      expectSnapshot(<Text subheading={2}>{HELLO_WORLD}</Text>);
    });

    it('should render as a p tag if the p prop is true or 1', () => {
      expectSnapshot(<Text p={true}>{HELLO_WORLD}</Text>);
      expectSnapshot(<Text p={1}>{HELLO_WORLD}</Text>);
    });

    it('should render as an aside if the p prop is set to 2', () => {
      expectSnapshot(<Text p={2}>{HELLO_WORLD}</Text>);
    });

    it('should render as a caption if the caption prop is enabled', () => {
      expectSnapshot(<Text caption={true}>{HELLO_WORLD}</Text>);
    });

    it('should render as the corresponding heading tag when the h prop is provided', () => {
      expectSnapshot(<Text h={1}>{HELLO_WORLD}</Text>);
      expectSnapshot(<Text h={2}>{HELLO_WORLD}</Text>);
      expectSnapshot(<Text h={3}>{HELLO_WORLD}</Text>);
      expectSnapshot(<Text h={4}>{HELLO_WORLD}</Text>);
      expectSnapshot(<Text h={5}>{HELLO_WORLD}</Text>);
      expectSnapshot(<Text h={6}>{HELLO_WORLD}</Text>);
    });

    it('should render as a span if a font weight is provided', () => {
      expectSnapshot(<Text weight="light">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text weight="regular">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text weight="medium">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text weight="semibold">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text weight="bold">{HELLO_WORLD}</Text>);
    });

    it('should render as a span if a text alignment is provided', () => {
      expectSnapshot(<Text align="left">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text align="center">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text align="right">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text align="inherit">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text align="initial">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text align="justify">{HELLO_WORLD}</Text>);
    });

    it('should render as a span if a text decoration is provided', () => {
      expectSnapshot(<Text decoration="line-through">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text decoration="overline">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text decoration="underline">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text decoration="none">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text decoration="inherit">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text decoration="initial">{HELLO_WORLD}</Text>);
    });

    it('should render as the provided type prop even if the display prop is provided', () => {
      expectSnapshot(<Text display={1} type="h1">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={1} type="h2">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={1} type="h3">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={1} type="h4">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={1} type="h5">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={1} type="h6">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={1} type="section">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={1} type="aside">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={1} type="caption">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={1} type="p">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={1} type="span">{HELLO_WORLD}</Text>);

      expectSnapshot(<Text display={2} type="h1">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={2} type="h2">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={2} type="h3">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={2} type="h4">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={2} type="h5">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={2} type="h6">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={2} type="section">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={2} type="aside">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={2} type="caption">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={2} type="p">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={2} type="span">{HELLO_WORLD}</Text>);

      expectSnapshot(<Text display={3} type="h1">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={3} type="h2">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={3} type="h3">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={3} type="h4">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={3} type="h5">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={3} type="h6">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={3} type="section">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={3} type="aside">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={3} type="caption">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={3} type="p">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={3} type="span">{HELLO_WORLD}</Text>);

      expectSnapshot(<Text display={4} type="h1">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={4} type="h2">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={4} type="h3">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={4} type="h4">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={4} type="h5">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={4} type="h6">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={4} type="section">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={4} type="aside">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={4} type="caption">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={4} type="p">{HELLO_WORLD}</Text>);
      expectSnapshot(<Text display={4} type="span">{HELLO_WORLD}</Text>);
    });
  });

  it('should provide the className if the children is a callback function', () => {
    expectSnapshot((
      <Text align="left">
        {({ className }) => <div className={className}>{HELLO_WORLD}</div>}
      </Text>
    ));
  });
});
