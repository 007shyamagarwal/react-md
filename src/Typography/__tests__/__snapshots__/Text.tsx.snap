// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text rendering return value should render a text string when there are no additional props provided 1`] = `"Hello, world!"`;

exports[`Text rendering return value should render a text string when there are no additional props provided 2`] = `"This is another string that is just some text. I'm not sure how helpful this is though."`;

exports[`Text rendering return value should render as a caption if the caption prop is enabled 1`] = `
<caption
  className="md-typography md-typography--caption"
>
  Hello, world!
</caption>
`;

exports[`Text rendering return value should render as a p tag if the p prop is true or 1 1`] = `
<p
  className="md-typography md-typography--body-1"
>
  Hello, world!
</p>
`;

exports[`Text rendering return value should render as a p tag if the p prop is true or 1 2`] = `
<p
  className="md-typography md-typography--body-1"
>
  Hello, world!
</p>
`;

exports[`Text rendering return value should render as a span if a font weight is provided 1`] = `
<span
  className="md-typography md-typography--light"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as a span if a font weight is provided 2`] = `
<span
  className="md-typography md-typography--regular"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as a span if a font weight is provided 3`] = `
<span
  className="md-typography md-typography--medium"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as a span if a font weight is provided 4`] = `
<span
  className="md-typography md-typography--semibold"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as a span if a font weight is provided 5`] = `
<span
  className="md-typography md-typography--bold"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as a span if a text alignment is provided 1`] = `
<span
  className="md-typography md-typography--left"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as a span if a text alignment is provided 2`] = `
<span
  className="md-typography md-typography--center"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as a span if a text alignment is provided 3`] = `
<span
  className="md-typography md-typography--right"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as a span if a text alignment is provided 4`] = `
<span
  className="md-typography md-typography--inherit"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as a span if a text alignment is provided 5`] = `
<span
  className="md-typography md-typography--initial"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as a span if a text alignment is provided 6`] = `
<span
  className="md-typography md-typography--justify"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as a span if a text decoration is provided 1`] = `
<span
  className="md-typography md-typography--line-through"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as a span if a text decoration is provided 2`] = `
<span
  className="md-typography md-typography--overline"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as a span if a text decoration is provided 3`] = `
<span
  className="md-typography md-typography--underline"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as a span if a text decoration is provided 4`] = `
<span
  className="md-typography md-typography--none"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as a span if a text decoration is provided 5`] = `
<span
  className="md-typography md-typography--inherit"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as a span if a text decoration is provided 6`] = `
<span
  className="md-typography md-typography--initial"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as an aside if the p prop is set to 2 1`] = `
<aside
  className="md-typography md-typography--body-2"
>
  Hello, world!
</aside>
`;

exports[`Text rendering return value should render as an h1 tag if the display prop is provided 1`] = `
<h1
  className="md-typography md-typography--display-1"
>
  Hello, world!
</h1>
`;

exports[`Text rendering return value should render as an h1 tag if the display prop is provided 2`] = `
<h1
  className="md-typography md-typography--display-2"
>
  Hello, world!
</h1>
`;

exports[`Text rendering return value should render as an h1 tag if the display prop is provided 3`] = `
<h1
  className="md-typography md-typography--display-3"
>
  Hello, world!
</h1>
`;

exports[`Text rendering return value should render as an h1 tag if the display prop is provided 4`] = `
<h1
  className="md-typography md-typography--display-4"
>
  Hello, world!
</h1>
`;

exports[`Text rendering return value should render as an h1 tag if the headline prop is enabled 1`] = `
<h1
  className="md-typography md-typography--headline"
>
  Hello, world!
</h1>
`;

exports[`Text rendering return value should render as an h3 tag if the subheading prop is set to 2 1`] = `
<h3
  className="md-typography md-typography--subheading-2"
>
  Hello, world!
</h3>
`;

exports[`Text rendering return value should render as an h4 tag if the subheading prop is enabled or set to 1 1`] = `
<h4
  className="md-typography md-typography--subheading-1"
>
  Hello, world!
</h4>
`;

exports[`Text rendering return value should render as an h4 tag if the subheading prop is enabled or set to 1 2`] = `
<h4
  className="md-typography md-typography--subheading-1"
>
  Hello, world!
</h4>
`;

exports[`Text rendering return value should render as the corresponding heading tag when the h prop is provided 1`] = `
<h1
  className="md-typography"
>
  Hello, world!
</h1>
`;

exports[`Text rendering return value should render as the corresponding heading tag when the h prop is provided 2`] = `
<h2
  className="md-typography"
>
  Hello, world!
</h2>
`;

exports[`Text rendering return value should render as the corresponding heading tag when the h prop is provided 3`] = `
<h3
  className="md-typography"
>
  Hello, world!
</h3>
`;

exports[`Text rendering return value should render as the corresponding heading tag when the h prop is provided 4`] = `
<h4
  className="md-typography"
>
  Hello, world!
</h4>
`;

exports[`Text rendering return value should render as the corresponding heading tag when the h prop is provided 5`] = `
<h5
  className="md-typography"
>
  Hello, world!
</h5>
`;

exports[`Text rendering return value should render as the corresponding heading tag when the h prop is provided 6`] = `
<h6
  className="md-typography"
>
  Hello, world!
</h6>
`;

exports[`Text rendering return value should render as the h2 tag if the title prop is enabled 1`] = `
<h2
  className="md-typography md-typography--title"
>
  Hello, world!
</h2>
`;

exports[`Text rendering return value should render as the provided type prop 1`] = `
<h1
  className="md-typography"
>
  Hello, world!
</h1>
`;

exports[`Text rendering return value should render as the provided type prop 2`] = `
<h2
  className="md-typography"
>
  Hello, world!
</h2>
`;

exports[`Text rendering return value should render as the provided type prop 3`] = `
<h3
  className="md-typography"
>
  Hello, world!
</h3>
`;

exports[`Text rendering return value should render as the provided type prop 4`] = `
<h4
  className="md-typography"
>
  Hello, world!
</h4>
`;

exports[`Text rendering return value should render as the provided type prop 5`] = `
<h5
  className="md-typography"
>
  Hello, world!
</h5>
`;

exports[`Text rendering return value should render as the provided type prop 6`] = `
<h6
  className="md-typography"
>
  Hello, world!
</h6>
`;

exports[`Text rendering return value should render as the provided type prop 7`] = `
<section
  className="md-typography"
>
  Hello, world!
</section>
`;

exports[`Text rendering return value should render as the provided type prop 8`] = `
<aside
  className="md-typography"
>
  Hello, world!
</aside>
`;

exports[`Text rendering return value should render as the provided type prop 9`] = `
<caption
  className="md-typography"
>
  Hello, world!
</caption>
`;

exports[`Text rendering return value should render as the provided type prop 10`] = `
<p
  className="md-typography"
>
  Hello, world!
</p>
`;

exports[`Text rendering return value should render as the provided type prop 11`] = `
<span
  className="md-typography"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 1`] = `
<h1
  className="md-typography md-typography--display-1"
>
  Hello, world!
</h1>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 2`] = `
<h2
  className="md-typography md-typography--display-1"
>
  Hello, world!
</h2>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 3`] = `
<h3
  className="md-typography md-typography--display-1"
>
  Hello, world!
</h3>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 4`] = `
<h4
  className="md-typography md-typography--display-1"
>
  Hello, world!
</h4>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 5`] = `
<h5
  className="md-typography md-typography--display-1"
>
  Hello, world!
</h5>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 6`] = `
<h6
  className="md-typography md-typography--display-1"
>
  Hello, world!
</h6>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 7`] = `
<section
  className="md-typography md-typography--display-1"
>
  Hello, world!
</section>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 8`] = `
<aside
  className="md-typography md-typography--display-1"
>
  Hello, world!
</aside>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 9`] = `
<caption
  className="md-typography md-typography--display-1"
>
  Hello, world!
</caption>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 10`] = `
<p
  className="md-typography md-typography--display-1"
>
  Hello, world!
</p>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 11`] = `
<span
  className="md-typography md-typography--display-1"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 12`] = `
<h1
  className="md-typography md-typography--display-2"
>
  Hello, world!
</h1>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 13`] = `
<h2
  className="md-typography md-typography--display-2"
>
  Hello, world!
</h2>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 14`] = `
<h3
  className="md-typography md-typography--display-2"
>
  Hello, world!
</h3>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 15`] = `
<h4
  className="md-typography md-typography--display-2"
>
  Hello, world!
</h4>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 16`] = `
<h5
  className="md-typography md-typography--display-2"
>
  Hello, world!
</h5>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 17`] = `
<h6
  className="md-typography md-typography--display-2"
>
  Hello, world!
</h6>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 18`] = `
<section
  className="md-typography md-typography--display-2"
>
  Hello, world!
</section>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 19`] = `
<aside
  className="md-typography md-typography--display-2"
>
  Hello, world!
</aside>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 20`] = `
<caption
  className="md-typography md-typography--display-2"
>
  Hello, world!
</caption>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 21`] = `
<p
  className="md-typography md-typography--display-2"
>
  Hello, world!
</p>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 22`] = `
<span
  className="md-typography md-typography--display-2"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 23`] = `
<h1
  className="md-typography md-typography--display-3"
>
  Hello, world!
</h1>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 24`] = `
<h2
  className="md-typography md-typography--display-3"
>
  Hello, world!
</h2>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 25`] = `
<h3
  className="md-typography md-typography--display-3"
>
  Hello, world!
</h3>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 26`] = `
<h4
  className="md-typography md-typography--display-3"
>
  Hello, world!
</h4>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 27`] = `
<h5
  className="md-typography md-typography--display-3"
>
  Hello, world!
</h5>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 28`] = `
<h6
  className="md-typography md-typography--display-3"
>
  Hello, world!
</h6>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 29`] = `
<section
  className="md-typography md-typography--display-3"
>
  Hello, world!
</section>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 30`] = `
<aside
  className="md-typography md-typography--display-3"
>
  Hello, world!
</aside>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 31`] = `
<caption
  className="md-typography md-typography--display-3"
>
  Hello, world!
</caption>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 32`] = `
<p
  className="md-typography md-typography--display-3"
>
  Hello, world!
</p>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 33`] = `
<span
  className="md-typography md-typography--display-3"
>
  Hello, world!
</span>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 34`] = `
<h1
  className="md-typography md-typography--display-4"
>
  Hello, world!
</h1>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 35`] = `
<h2
  className="md-typography md-typography--display-4"
>
  Hello, world!
</h2>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 36`] = `
<h3
  className="md-typography md-typography--display-4"
>
  Hello, world!
</h3>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 37`] = `
<h4
  className="md-typography md-typography--display-4"
>
  Hello, world!
</h4>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 38`] = `
<h5
  className="md-typography md-typography--display-4"
>
  Hello, world!
</h5>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 39`] = `
<h6
  className="md-typography md-typography--display-4"
>
  Hello, world!
</h6>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 40`] = `
<section
  className="md-typography md-typography--display-4"
>
  Hello, world!
</section>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 41`] = `
<aside
  className="md-typography md-typography--display-4"
>
  Hello, world!
</aside>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 42`] = `
<caption
  className="md-typography md-typography--display-4"
>
  Hello, world!
</caption>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 43`] = `
<p
  className="md-typography md-typography--display-4"
>
  Hello, world!
</p>
`;

exports[`Text rendering return value should render as the provided type prop even if the display prop is provided 44`] = `
<span
  className="md-typography md-typography--display-4"
>
  Hello, world!
</span>
`;

exports[`Text should provide the className if the children is a callback function 1`] = `
<div
  className="md-typography md-typography--left"
>
  Hello, world!
</div>
`;
